### DISCUSSION TITLE
Criteria APIのログ出力のバグ？

### DISCUSSION THREAD

**Discussion by gikeihi** [domaframework/doma#893] [Q&A] [answered]
環境：
DB: MySQL 5.7.34
Doma2:  2.53.1

テーブルのColumnをtinyint(1)を定義すると、domaのcodegenを利用して、Booleanのフィルドを生成します。
Criteria APIを利用して、SQLログを出す時、`true`や`false`はsingle quotationで囲まれてしまいました。
例：
```
t0_.use_flag = 'true'
```
アプリ自体の動きは問題ございませんが、SQLログだけは上記のようになります。
SQLログを使って、データの抽出を検証する時、single quotationを一旦除外しないといけないですので、ちょっと面倒です。

これはバグかどうか、ご確認いただけませんか？

**Comment by nakamura-to** [comment, marked as answer]
共有ありがとうございます。
SQLの真偽値のリテラルをどう表現するかはDBMSごとに違いがあります。例えばSQL Serverではsingle quotationで囲む必要があるのですがMySQLでは異なるようですね。次のバージョンではMySQLの場合は囲まないように修正しようと思います。

ちなみに、2.53.1でも下記リンク先のメソッドをオーバーライドするようなコードを書くことで対応ができます。
https://github.com/domaframework/doma/blob/2.53.1/doma-core/src/main/java/org/seasar/doma/jdbc/dialect/StandardDialect.java#L591

（MysqlSqlLogFormattingVisitorを継承したクラスでオーバーライドし、MysqlSqlLogFormattingVisitorのインスタンスをMysqlDialectのコンストラクタに渡すようにします）

**Reply by gikeihi** [reply]
ご回答ありがとうございます。

> SQLの真偽値のリテラルをどう表現するかはDBMSごとに違いがあります。
仰った通り、違いますよね、MySQLの場合、囲まれると、全て`false`となります。

ご教授のより、暫定案として、`MysqlSqlLogFormattingVisitor `をカスタマイズします。
